### DISCUSSION TITLE
Phase detection for Playwright tests

### DISCUSSION THREAD

**Discussion by matchatype** [vercel/next.js#34825] [Help] [answered]
Playwright can spin up an instance of next dev server automatically before running tests by passing the `webServer` config object. Additionally, Playwright allows to capture full screen screenshots for visual diffing. I noticed such screenshots to be a bit flaky, because at times Next's dev indicator gets in the way, dirtying up the final screenshot.

I know I can disable the indicator by passing a `devIndicators` object to next.config.js and I know I can selectively pass different configuration objects based on phases. What I couldn't figure out is a way to keep devIndicators during development and disabling it when next is run by Playwrights. Any hints?

**Comment by wwwdepot** [comment, marked as answer]
In playwright webServer config, you can set environment variables there. Use this env var in next.config.js to disable `devIndicators`.
